src/createPackage/models/createPackageManager.ts

```diff
@@ -165,6 +191,132 @@ export class CreatePackageManager {
+    const tileEstimatedSize = this.getTileEstimatedSize(layerMetadata.tileOutputFormat as TileOutputFormat);
+
+    resourceId = resourceId as string;
+    version = version as string;
+    productType = productType as ProductType;
+
+    const srcRes = layerMetadata.maxResolutionDeg as number;
+    const maxZoom = degreesPerPixelToZoomLevel(srcRes);
+
+    // ROI vs layer validation section - zoom + geo intersection
+    featuresRecords.forEach((record) => {
+      if (record.zoomLevel > maxZoom) {
+        throw new BadRequestError(`The requested resolution ${record.targetResolution} is larger then then product resolution ${srcRes}`);
+      }
+
+      // generate sanitized bbox for each original feature
+      record.sanitizedBox = this.sanitizeBbox(
+        record.geometry as Polygon | MultiPolygon,
+        layerMetadata.footprint as Polygon | MultiPolygon,
+        record.zoomLevel
+      );
+      if (!record.sanitizedBox) {
+        throw new BadRequestError(
+          `Requested ${JSON.stringify(record.geometry as Polygon | MultiPolygon)} has no intersection with requested layer ${
+            layer.metadata.id as string
+          }`
+        );
+      }
+    });
+    const layerBbox = PolygonBbox(roi); // bounding box of entire ROI
+
+    const dupParams: JobExportDuplicationParams = {
+      resourceId,
+      version,
+      dbId,
+      roi,
+      crs: crs ?? DEFAULT_CRS,
+    };
+
+    const callbacks = callbackURLs.map((url) => <ICallbackTargetExport>{ url, roi });
+    const duplicationExist = await this.checkForExportDuplicate(dupParams, callbacks);
+    if (duplicationExist && duplicationExist.status === OperationStatus.COMPLETED) {
+      return duplicationExist;
+    } else if (duplicationExist) {
+      return duplicationExist;
+    }
+
+    const batches: ITileRange[] = [];
+    // const batches = this.generateTileGroups(polygon as Polygon, layerMetadata.footprint as Polygon | MultiPolygon, zoomLevel);
+    featuresRecords.forEach((record) => {
+      // for (let i = 0; i <= record.zoomLevel; i++) {
+      // const recordBatches = bboxToTileRange(record.sanitizedBox as BBox2d, i);
+      const recordBatches = this.generateTileGroups(
+        record.geometry as Polygon | MultiPolygon,
+        layerMetadata.footprint as Polygon | MultiPolygon,
+        record.zoomLevel
+      );
+      // batches.push(recordBatches);
+      batches.push(...recordBatches);
+      // }
+    });
+
+    const estimatesGpkgSize = calculateEstimateGpkgSize(batches, tileEstimatedSize); // size of requested gpkg export
+    if (this.storageEstimation.validateStorageSize) {
+      const isEnoughStorage = await this.validateFreeSpace(estimatesGpkgSize);
+      if (!isEnoughStorage) {
+        throw new InsufficientStorage(`There isn't enough free disk space to executing export`);
+      }
+    }
+    const separator = this.getSeparator();
+    const packageName = this.generateExportPackageName(productType, resourceId, version, featuresRecords);
```

Comment:
why don't you return the template withaot the suffix, and the just add the .gpkg or .json?

Reply:
done